Foundation/NSHost.swift

```diff
@@ -78,11 +113,19 @@ open class Host: NSObject {
             hints.ai_flags = flags
             
             var res0: UnsafeMutablePointer<addrinfo>? = nil
+            defer {
+                freeaddrinfo(res0)
```

Comment:
this defer should be after the `getaddrinfo` because if it fails the res0 might not be initialized by the function.